### PR TITLE
Add functional test for secrets scenario 1 and 4

### PR DESCRIPTION
**Pull request by minz1027**
- Add cucumber test for secrets.feature scenario 1 and 4 (need more users for other scenarios)
- In scenario 1, remove `And the build succeeded` since it is already checked in `Then the "foo" secret should be available in the build`
- PipelineId is hardcoded until we find a better way to do it
- [Repo used for testing](https://github.com/screwdriver-cd-test/functional-secrets)

issue [#148](https://github.com/screwdriver-cd/screwdriver/issues/148)

### CODE REVIEW

File: features/step_definitions/secrets.js [hunk 1]
```diff
@@ -0,0 +1,128 @@
+'use strict';
+const Assert = require('chai').assert;
+const request = require('../support/request');
+
+module.exports = function server() {
+    this.setDefaultTimeout(60000);
```
Comment (FenrirUnbound) [inline features/step_definitions/secrets.js]: This affects timeouts for ***all*** tests

File: features/support/before-hook.js [hunk 2]
```diff
@@ -10,6 +23,18 @@ function beforeHooks() {
...
+        this.namespace = 'v4';
+        this.waitForBuild = (buildID) =>
+            requestretry({
+                uri: `${this.instance}/${this.namespace}/builds/${buildID}`,
+                method: 'GET',
+                maxAttempts: 10,
+                retryDelay: 5000,
+                retryStrategy: buildRetryStrategy,
+                json: true
+            });
```
Comment (tkyi) [inline features/support/before-hook.js]: 👍 nice!

File: features/step_definitions/git-flow.js [hunk 3]
```diff
@@ -157,7 +157,7 @@ function waitForBuild(screwdriverInstance, desiredSha) {
 module.exports = function server() {
     // eslint-disable-next-line new-cap
     this.Before(() => {
-        this.instance = 'http://api.screwdriver.cd';
```
Comment (FenrirUnbound) [inline features/step_definitions/git-flow.js]: No need to modify this feature set

Comment (d2lam) [inline features/step_definitions/git-flow.js]: Had to. Its `Before` applies to all the scenarios. We could move this whole block to `before-hooks` but didn't want to mess with git-flow

File: features/step_definitions/secrets.js (hunk 1 above)
Comment (d2lam) [inline features/step_definitions/secrets.js]: Thanks for telling me how to fix this 👍

File: features/step_definitions/git-flow.js (hunk 3 above)
Comment (FenrirUnbound) [inline features/step_definitions/git-flow.js]: 👍

**Pull request merged**